server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/LogicalPlanVisitor.java

```diff
@@ -232,12 +237,34 @@ public PlanNode visitQueryBody(
           }
         }
 
-        planBuilder =
-            planBuilder.planTransform(
-                transformExpressions,
-                queryStatement.isGroupByTime(),
-                queryStatement.getSelectComponent().getZoneId(),
-                queryStatement.getResultTimeOrder());
+        if (queryStatement.isGroupByLevel()) {
+          if (havingExpression != null) {
```

Comment:
We don't need this judgment here because this judgment will be made in planFilterAndTransform.